File: PersonalAssistantUI/src/app/models/room/room.ts
import { User } from "..";

export class Room {
    public id: number;
    public name: string;
    public icon: string; 
    public conversationType: string;
    public roomType: string;
    public password: string;
    public hasJoined: boolean;
    public scope: string;
    public ownerId: number;
    public owner: User;
    public createdAt: Date;
    public updatedAt: Date;
    public membersCount: number;
    public members: User[]

    constructor(name: string, type: string, conversationType: string, ownerId: number, password: string = null) {
        this.ownerId = ownerId;
        this.name = name;
        this.roomType = type;
        this.conversationType = conversationType;
        this.password = password;        
    }
}

File: PersonalAssistant.DataAccess/Entities/Room.cs
using System;
using System.Collections.Generic;
using PersonalAssistant.DataAccess.Identity;

namespace PersonalAssistant.DataAccess.Entities
{
    public class Room
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Icon { get; set; }
        public string ConversationType { get; set; }
        public string RoomType { get; set; }
        public string Password { get; set; }
        public int? OwnerId { get; set; }
        public User Owner { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public IEnumerable<UserRoom> UserRooms { get; set; }
        public IEnumerable<Message> Messages { get; set; }
    }
}
